wordpress-mobile/WordPress-Android
# [SotW 2023] Add State of the Word post-event card to WordPress

Fixes #19709

More info in the parent issue. The card is only visible for the following requirements:
- The app is WordPress (part of the `WpSotw2023NudgeFeatureConfig` logic)
- The `wp_sotw_2023_nudge` flag is ON (`WpSotw2023NudgeFeatureConfig`)
- The user did not hide the card (App Pref set by user interaction with `hide this` card button)
- The device's current date is at least one day after the event date (2023-12-11)
- The user app language is English (`en`)


https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/c976d05a-7b33-4f42-964b-7b12de4e8045

-----

## To Test:

Note: tracking is sent for all card interactions, check the analytics PR (https://github.com/wordpress-mobile/WordPress-Android/pull/19719) for testing instructions.

1. Install and log into the WordPress app
3. Set up app/device to follow the requirements above (some tips: https://github.com/wordpress-mobile/WordPress-Android/pull/19718#issue-2021569561)
4. Go to `My Site`
5. **Verify** the SotW card is shown above the Site Menu items
6. Tap `Watch Now`
7. **Verify** the SotW 2023 event page is opened on an external browser
8. Go back
9. Tap `Hde this`
10. **Verify** the card is hidden from the dashboard (all sites)

Note: the card only shows again if the user reinstalls the app, clears app data, or log out and log in again.

-----

## Regression Notes

1. Potential unintended areas of impact

    - Wrong behavior in the site dashboard
    - Card showing in Jetpack

12. What I did to test those areas of impact (or what existing automated tests I relied on)

    - Manual tests and Unit tests

13. What automated tests I added (or what prevented me from doing so)

    - Unit tests for all new code introduced
    - Updated unit tests for code that was changed

-----

## PR Submission Checklist:


-----

## UI Changes Testing Checklist:
